include/zephyr/toolchain/gcc.h

```diff
@@ -267,6 +267,10 @@ do {                                                                    \
  */
 #endif
 
+#ifndef __deprecated_version
+#define __deprecated_version(version)	__attribute__((deprecated(#version)))
+#endif
```

Comment:
We need to ensure that this macro is available in every toolchain supported (e.g. `iccarm.h` included by `iar.h`); otherwise, code using this macro will fail to compile on the toolchains missing this macro.